msal/src/internal/java/com/microsoft/identity/client/AuthenticationRequestParameters.java

```diff
@@ -70,6 +71,12 @@ private AuthenticationRequestParameters(final Authority authority, final TokenCa
         mClientId = clientId;
         mSliceParameters = sliceParameters;
         mRequestContext = requestContext;
+
+        if (authority.getAuthorityType() == Authority.AuthorityType.B2C) {
+            mWebViewSelection = WebViewSelection.SYSTEM_BROWSER;
+        } else {
+            mWebViewSelection = WebViewSelection.EMBEDDED_WEBVIEW;
```

Comment:
nit: initialize the variable with EMBEDDED_WEBVIEW and then you don't need the else block